src/lib/services/index.ts

```diff
@@ -362,9 +362,17 @@ export const createServices = (
         config.getLogger,
     );
 
-    const { featureLifecycleService } = db
-        ? createFeatureLifecycleService(db, config)
-        : createFakeFeatureLifecycleService(config);
+    const transactionalFeatureLifecycleService = db
+        ? withTransactional(
+              (db: Db) =>
+                  createFeatureLifecycleService(db, config)
```

Comment:
this is slightly different from other code because createFeatureLifecycleService composition root returns more than a service